java-source/src/main/java/com/example/flow/ExampleFlow.java

```diff
@@ -135,84 +135,67 @@ public Initiator(IOUState iou, Party otherParty) {
-                return new ExampleFlowResult.Failure(ex.getMessage());
-            }
+        public Acceptor(Party otherParty) {
+            this.otherParty = otherParty;
         }
-    }
-
-    public static class ExampleFlowResult {
-        public static class Success extends com.example.flow.ExampleFlow.ExampleFlowResult {
-            private String message;
 
-            private Success(String message) { this.message = message; }
-
-            @Override
-            public String toString() { return String.format("Success(%s)", message); }
+        @Override
+        public ProgressTracker getProgressTracker() {
+            return progressTracker;
         }
 
-        public static class Failure extends com.example.flow.ExampleFlow.ExampleFlowResult {
-            private String message;
-
-            private Failure(String message) { this.message = message; }
-
-            @Override
-            public String toString() { return String.format("Failure(%s)", message); }
+        @Suspendable
+        @Override
+        public SignedTransaction call() throws FlowException {
+            // Prep.
+            // Obtain a reference to our key pair.
+            final KeyPair keyPair = getServiceHub().getLegalIdentityKey();
+            final Party notary = single(getServiceHub().getNetworkMapCache().getNotaryNodes()).getNotaryIdentity();
+            // Obtain a reference to the notary we want to use and its public key.
+            final CompositeKey notaryPubKey = notary.getOwningKey();
+
+            // Stage 5.
+            progressTracker.setCurrentStep(RECEIVING_TRANSACTION);
+            // All messages come off the wire as UntrustworthyData. You need to 'unwrap' them. This is where you
+            // validate what you have just received.
+
+            final SignedTransaction partSignedTx = receive(SignedTransaction.class, otherParty)
+                    .unwrap(tx ->
+                    {
+                        // Stage 6.
+                        progressTracker.setCurrentStep(VERIFYING_TRANSACTION);
+                        try {
+                            // Check that the signature of the other party is valid.
+                            // Our signature and the notary's signature are allowed to be omitted at this stage as
+                            // this is only a partially signed transaction.
+                            final WireTransaction wireTx = tx.verifySignatures(CryptoUtilities.getComposite(keyPair.getPublic()), notaryPubKey);
+
+                            // Run the contract's verify function.
+                            // We want to be sure that the agreed-upon IOU is valid under the rules of the contract.
+                            // To do this we need to run the contract's verify() function.
+                            wireTx.toLedgerTransaction(getServiceHub()).verify();
+                        } catch (SignatureException|TransactionResolutionException ex) {
```

Comment:
`unwrap` can throw FlowException so we don't need to catch TransactionResolutionException, but we do for SignatureException. For this we can throw a FlowException instead of RuntimeException with the relevant message.